src/plans/stopping_criterion.jl

```diff
@@ -623,6 +623,84 @@ function update_stopping_criterion!(c::StopWhenStepsizeLess, ::Val{:MinStepsize}
     return c
 end
 
+"""
+    StopWhenCostNan <: StoppingCriterion
+
+stop looking at the cost function of the optimization problem from within a [`AbstractManoptProblem`](@ref), i.e `get_cost(p,get_iterate(o))`.
+
+# Constructor
+
+    StopWhenCostNan()
+
+initialize the stopping criterion to NaN.
+"""
+mutable struct StopWhenCostNan <: StoppingCriterion
+    reason::String
+    at_iteration::Int
+    StopWhenCostNan() = new("", 0)
+end
+function (c::StopWhenCostNan)(
+    p::AbstractManoptProblem, s::AbstractManoptSolverState, i::Int
+)
+    if i == 0 # reset on init
+        c.reason = ""
+        c.at_iteration = 0
+    end
+    if i > 0 && isnan(get_cost(p, get_iterate(s)))
+        c.reason = "The algorithm reached a cost function value ($(get_cost(p,get_iterate(s)))).\n"
+        c.at_iteration = 0
+        return true
+    end
+    return false
+end
+function status_summary(c::StopWhenCostNan)
+    has_stopped = length(c.reason) > 0
+    s = has_stopped ? "reached" : "not reached"
+    return "f(x) is NaN:\t$s"
+end
+function show(io::IO, c::StopWhenCostNan)
+    return print(io, "StopWhenCostNan()\n    $(status_summary(c))")
+end
+
+"""
+    StopWhenIterNan <: StoppingCriterion
```

Comment:
I remember implementing it because, iirc, in using the ALM for some estimations I was carrying out this would sometimes happen, but I wouldn't be able to reproduce it immediately. I will get back to you.

Reply:
Ah! Looking at the code closer (before I wanted to delete it), the name was a bit ambiguous.

I thought you were speaking about the `Iteration` ($k=1,2,3,4,...$) until that hits NaN the algorithms have to run very long.
But what is. actually meant here is the `Iterate` (usually `state.p`). And sure for debugging reasons we can keep that as well.